src/EditorFeatures/Core/SolutionEvents/HostSolutionEventsWorkspaceEventListener.cs

```diff
@@ -0,0 +1,305 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.SolutionEvents
+{
+    [ExportEventListener(WellKnownEventListeners.Workspace, WorkspaceKind.Host), Shared]
+    internal sealed partial class HostSolutionEventsWorkspaceEventListener : IEventListener<object>
```

Comment:
this is the part that runs in VS and connects to the actual VS workspace and hears about changes to it.  It is very similar in that regard to the equivalent HostSolutionCrawlerWorkspaceEventListener type.